zuul-core/src/main/java/com/netflix/zuul/netty/timeouts/OriginTimeoutManager.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright 2021 Netflix, Inc.
+ *
+ *      Licensed under the Apache License, Version 2.0 (the "License");
+ *      you may not use this file except in compliance with the License.
+ *      You may obtain a copy of the License at
+ *
+ *          http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *      Unless required by applicable law or agreed to in writing, software
+ *      distributed under the License is distributed on an "AS IS" BASIS,
+ *      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *      See the License for the specific language governing permissions and
+ *      limitations under the License.
+ */
+
+package com.netflix.zuul.netty.timeouts;
+
+import static com.netflix.client.config.CommonClientConfigKey.ReadTimeout;
+
+import com.google.common.base.Strings;
+import com.netflix.client.config.DefaultClientConfigImpl;
+import com.netflix.client.config.IClientConfig;
+import com.netflix.config.DynamicLongProperty;
+import com.netflix.zuul.context.CommonContextKeys;
+import com.netflix.zuul.message.http.HttpRequestMessage;
+import com.netflix.zuul.origins.NettyOrigin;
+import java.time.Duration;
+import javax.annotation.Nullable;
+
+/**
+ * Origin Timeout Manager
+ *
+ * @author Arthur Gonigberg
+ * @since February 24, 2021
+ */
+public class OriginTimeoutManager {
```

Comment:
final

Reply:
This class can be overridden in the same way that `NettyOrigin` can.